tests/pytest/test_semantic_groupby.py

```diff
@@ -0,0 +1,127 @@
+#!/usr/bin/env python3
+"""
+Test script for semantic group by operation.
+
+This script tests the SemanticGroupByOp implementation by creating a small dataset
+of product reviews and grouping them by complaint type.
+"""
+
+import pandas as pd
+import palimpzest as pz
+from palimpzest.query.operators.aggregate import SemanticGroupByOp
+from palimpzest.constants import Model
+
+# Define columns for the review schema
+review_cols = [
+    {"name": "complaint", "type": str, "desc": "The type of complaint mentioned in the review (e.g., size, quality, shipping, description mismatch, ergonomics)"},
+]
+
+def test_semantic_groupby_basic():
+    """Test basic semantic group by functionality using the physical operator directly."""
+    print("Testing SemanticGroupByOp basic functionality...")
+    
+    try:
+        # Create list of candidates from text file dataset with schema
+        ds = pz.TextFileDataset(id="reviews", path="product-reviews/")
+        ds = ds.sem_map(review_cols)  # Add schema to extract complaint types
```

Comment:
You shouldn't need to use a `sem_map` to extract the `complaint` field. My guess is you needed this before because of the validation check for `gby_fields` in the `input_schema`, but if you remove that check then the test example I shared in the Github issue should work without the `sem_map`